### ISSUE TITLE
0.2.2-sandbox+oryOS.10: Unable to initialize compiler: lstat /go: no such file or directory

### ISSUE THREAD

**Issue by sevenlure** [ory/keto#73]
**Describe the bug**
Version [0.2.2-sandbox+oryOS.10]
keto serve
=> Unable to initialize compiler: lstat /go: no such file or directory

Version [v0.1.9-sandbox+oryOS.9]
keto serve
=> work well

**To Reproduce**
1. Run `keto serve`
2. See error: Unable to initialize compiler: lstat /go: no such file or directory

**Version:**
 - Environment: [binaries]
 - Version [0.2.2-sandbox+oryOS.10]

**Comment by bensont1** [comment]
I have also run into this problem, same issue on `0.2.1-sandbox+oryOS.10`.

**Comment by aeneasr** [comment]
What OS?

**Comment by bensont1** [comment]
I’m runnning the Docker image from DockerHub

**Comment by bensont1** [comment]
I think i found the issue, seems to be in this block on HTTPS server start. The certs are not actually set in the http.Server tls config

https://github.com/ory/keto/blob/ed7af3fa4e5d1d0d03b5366f4cf865a5b82ec293/cmd/server/serve.go#L126-L130

**Comment by aeneasr** [comment]
Are you running it with TLS configured or without?

**Comment by bensont1** [comment]
I am running `0.2.2-sandbox+oryOS.10` with TLS configured using Path to cert and key.

**Comment by aeneasr** [comment]
Can you make sure that the paths are set correctly? Could you maybe show the env config?

**Comment by aeneasr** [comment]
Right, the logic here seems to be messed up. TLSConfig should be nil in case of `tlsx.ErrNoCertificatesConfigured` and set if the error is nil.

**Comment by sum2000** [comment]
@aeneasr  I am facing the same issue on Mac. I am building from source and I pull in the src using `go mod vendor`. This script pulls in Keto source (.go files) to local vendor folder and I build using 'go build -mod=vendor'. The build is fine but when I run `keto serve`, I get the same error as OP. It has to do with `go mod vendor` because it does not pull the `.rego` files from git as they are non .go files. I manually added the `rego` folder on my local src copy inside vendor and everything was working fine. One solution could be adding a placeholder `.go` file inside the `rego` folder

**Comment by aeneasr** [comment]
Ohhh, that's possible. Damn, any ideas how to fix that?

**Comment by sum2000** [comment]
One solution could be adding a placeholder .go file inside the rego folder. I imagine this will force go mod vendor to download all the necessary code required to run keto serve

**Comment by sum2000** [comment]
@aeneasr  [PR](https://github.com/ory/keto/pull/76) for the issue

**Comment by aeneasr** [comment]
Whoops, reopening because this is two issues in one

**Comment by ingenieroariel** [comment]
I got a similar error with yesterday's master:
```
Unable to initialize compiler: lstat /build: no such file or directory
```

**Comment by aeneasr** [comment]
This is probably caused by: https://github.com/ory/keto/pull/79#issuecomment-459759675

**Comment by ingenieroariel** [comment]
I have a nix derivation that reproduces the problem (while packaging ORY for NixOS).

https://github.com/piensa/nur-packages/blob/master/pkgs/keto/default.nix

**Comment by aeneasr** [comment]
Ok, it seems like there were several issues:

- The CircleCI build spec did not use packr to pack the rego files in the binary
- The makefile did not use packr to pack the rego files in the binary
- The docs did not specify that packr needs to be installed

This has all been fixed now and is being merged with #84
